src/modules/job-exec/bulk-exec.c

```diff
@@ -222,11 +222,20 @@ static void exec_output_cb (flux_subprocess_t *p, const char *stream)
 {
     struct bulk_exec *exec = flux_subprocess_aux_get (p, "job-exec::exec");
     const char *s;
+    int flags = flux_subprocess_get_flags (p);
     int len;
 
-    if ((len = flux_subprocess_getline (p, stream, &s)) < 0) {
-        flux_log_error (exec->h, "flux_subprocess_getline");
-        return;
+    if (flags & FLUX_SUBPROCESS_FLAGS_LOCAL_UNBUF) {
+        if ((len = flux_subprocess_read_local_unbuf (p, stream, &s)) < 0) {
+            flux_log_error (exec->h, "flux_subprocess_read_local_unbuf");
+            return;
```

Comment:
I think it would be fine to assume the UNBUF flag, e.g. just |= to the flags provided  to` bulk_exec_push_cmd()` and drop the new public function.

Reply:
sounds good